File: src/main/java/io/github/dqualizer/dqlang/types/rqa/definition/Artifact.java
package io.github.dqualizer.dqlang.types.rqa.definition;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Artifact {


    @JsonProperty("system_id")
    private String systemId;


    @JsonProperty("activity_id")
    private String activityId;


    public boolean isNode() {
        return systemId != null;
    }

    public boolean isEdge() {
        return systemId == null;
    }
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/rqa/definition/Artifact.kt
package io.github.dqualizer.dqlang.types.rqa.definition

import com.fasterxml.jackson.annotation.JsonProperty

data class Artifact(
  @JsonProperty("system_id")
  val systemId: String? = null,

  @JsonProperty("activity_id")
  val activityId: String? = null
) {
  fun isNode(): Boolean = systemId != null

  fun isEdge(): Boolean = systemId == null
}
